raiden-ts/src/transfers/epics.ts

```diff
@@ -974,16 +1092,53 @@ export const transferChannelClosedEpic = (
   );
 
 /**
- * Sends Processed for unhandled transfer messages
+ * Receiving RefundTransfer for pending transfer fails it
+ *
+ * @param action$  Observable of messageReceived actions
+ * @param state$  Observable of RaidenStates
+ * @returns  Observable of transferFailed|transferRefunded actions
+ */
+export const transferRefundedEpic = (
+  action$: Observable<RaidenAction>,
+  state$: Observable<RaidenState>,
+): Observable<ActionType<typeof transferRefunded | typeof transferFailed>> =>
+  action$.pipe(
+    filter(isActionOf(messageReceived)),
+    withLatestFrom(state$),
+    mergeMap(function*([action, state]) {
+      const message = action.payload.message;
+      if (!message || !Signed(RefundTransfer).is(message)) return;
+      const secrethash = message.lock.secrethash;
```

Comment:
We could destructure here 

- `action.payload.message`
- `message.lock.secrethash`
- `state.sent[secrethash].transfer[1]`

might improve readability of the if condition

Reply:
This would require splitting the `if`, because e.g. `sent[secrethash]` isn't defined if `!(secrethash in state.sent)`, and so on. Can you suggest code on how you think it'd be better, please?

Reply:
I think splitting up the if might actually be good as we could do a short comment on why we return there (besides that we already have another if above)

I thought about something like this:
```
const sentTransfer = state.sent[secrethash];
if (!sentTransfer) return;

const [_, firstTransfer] = sentTransfer.transfer,
  {
    recipient,
    payment_identifier,
    lock: { amount, expiration },
  } = firstTransfer;

if (
  message.initiator !== recipient ||
  !message.payment_identifier.eq(payment_identifier) ||
  !message.lock.amount.eq(amount) ||
  !message.lock.expiration.eq(expiration) ||
  sentTransfer.unlock || // already unlocked
  sentTransfer.lockExpired || // already expired
  sentTransfer.channelClosed || // channel closed
  message.lock.expiration.lte(state.blockNumber) // lock expired but transfer didn't yet
)
  return;
```

But it's mostly about readability so no big deal

Reply:
Did almost what you suggested, just not the destructuring of the first transfer. Thank you